torch/_utils.py

```diff
@@ -577,14 +577,27 @@ def _get_devices_properties(device_ids):
     return [_get_device_attr(lambda m: m.get_device_properties(i)) for i in device_ids]
 
 
+# can be used in JIT script.
+def is_backend_available(device_type: str) -> bool:
+    if device_type == "cuda":
+        return torch._C._is_hooks_available("cuda") and torch.cuda.device_count() > 0
+    if device_type == "xpu":
+        return torch._C._is_hooks_available("xpu") and torch.xpu.device_count() > 0  # type: ignore[attr-defined]
```

Comment:
Hi @bdhirsh, I encounter a dilemma that torch has no attribute 'xpu' in TorchScript here. If CUDA is disabled, torch.cuda.device_count() will not be supported in TorchScript, and vice versa, torch.xpu.device_count() will not be supported when XPU is disabled.